scipy/stats/tests/test_continuous.py

```diff
@@ -1560,17 +1556,68 @@ def test_generate_domain_support(self):
         msg = _generate_domain_support(_LogUniform)
         assert "accepts two parameterizations" in msg
 
-    def test_ContinuousDistribution__str__(self):
+    def test_ContinuousDistribution__repr__(self):
         X = _Uniform(a=0, b=1)
-        assert str(X) == "_Uniform(a=0.0, b=1.0)"
+        assert repr(X) == "_Uniform(a=0.0, b=1.0)"
 
-        assert str(X*3 + 2) == "ShiftedScaled_Uniform(a=0.0, b=1.0, loc=2.0, scale=3.0)"
+        assert repr(X*3 + 2) == "3.0*_Uniform(a=0.0, b=1.0) + 2.0"
 
         X = _Uniform(a=np.zeros(4), b=1)
-        assert str(X) == "_Uniform(a, b, shape=(4,))"
+        assert repr(X) == "_Uniform(a, b, shape=(4,))"
 
         X = _Uniform(a=np.zeros(4, dtype=np.float32), b=np.ones(4, dtype=np.float32))
-        assert str(X) == "_Uniform(a, b, shape=(4,), dtype=float32)"
+        assert repr(X) == "_Uniform(a, b, shape=(4,), dtype=float32)"
+
+
+
+class TestReprs:
+    U = _Uniform(a=0, b=1)
+    X = Normal(mu=-1, sigma=1)
+    Y = Normal(mu=1, sigma=1)
+
+    def _get_fingerprint(self, dist):
+        # Get a sample from a distribution or array of distributions for use
+        # as a fingerprint for loosely testing equality of distributions
+        # or arrays of distributions.
+        dist = np.asarray(dist)
+        if dist.ndim:
+            return np.fromiter(
+                (val for x in np.asarray(dist) for val in x.sample(shape=10, rng=1234)),
+                dtype=float)
+        return dist[()].sample(shape=10, rng=1234)
+
+    @pytest.mark.parametrize(
+        "dist",
+        [
+            U,
+            3*U + 2,
+            U**4,
+            (3*U + 2)**4,
+            (3*U + 2)**3,
+            2**U,
+            2**(3*U + 1),
+            1 / (1 + U),
+            stats.order_statistic(U, r=3, n=5),
+            stats.truncate(U, 0.2, 0.8),
+            stats.Mixture([X, Y], weights=[0.3, 0.7]),
+            abs(U),
+            stats.exp(U),
+            stats.log(1 + U),
+            np.array([1.0, 2.0])*U + np.array([2.0, 3.0]),
```

Comment:
Ah, I recently noticed this while working on `marray` but I see it applies here, too.
NumPy arrays define `__mul__`, and the result of this operation is:

```python3
np.array([1.0, 2.0])*U + np.array([2.0, 3.0]),
# array([1.0*_Uniform(a=0.0, b=1.0) + 2.0, 2.0*_Uniform(a=0.0, b=1.0) + 3.0],
#       dtype=object)
```

That's unfortunate, because it should really work like:
```python3
U*np.array([1.0, 2.0]) + np.array([2.0, 3.0]),
# ([1. 2.]*_Uniform(a=0.0, b=1.0) + [2. 3.],)
```
In this test, I'd suggest doing the latter - although something needs adjustment because the arrays look like they have been rendered with `__str__` rather than `__repr__`. 

When that's done, rather than having a separate `_get_fingerprint`, you should just be able to call `sample`.

Fixing the asymmetry of array operators is a different story...

Reply:
Done, this worked.